src/Utilities/StdHelpers.hpp

```diff
@@ -283,3 +284,82 @@ std::string formatted_string(const std::string& fmt, Args... args) {
   }
   return rtn;
 }
+
+// Arithmetic operators for std::array<T, Dim>
+
+template <size_t Dim, typename T>
+inline std::array<T, Dim>& operator+=(
+    std::array<T, Dim>& lhs,
+    const std::array<T, Dim>& rhs) noexcept(noexcept(lhs[0] += rhs[0])) {
+  for (size_t i = 0; i < Dim; ++i) {
+    gsl::at(lhs, i) += gsl::at(rhs, i);
+  }
+  return lhs;
+}
+
+// clang-tidy: Do not use const_cast
+template <size_t Dim, typename T>
+inline std::array<T, Dim> operator+(
+    const std::array<T, Dim>& lhs,
+    const std::array<T, Dim>&
+        rhs) noexcept(noexcept(const_cast<std::array<T, // NOLINT
```

Comment:
Couldn't this be `std::declval<std::array<T, Dim>&>()` and avoid the clang-tidy complaints?

Reply:
Yes it can be. Will fix.